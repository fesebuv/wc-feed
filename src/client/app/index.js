const { default: Vue }  = require('vue');

Vue.component('image-item', {
  props: ['item'],
  template: '<img v-bind:src="item.src" v-bind:alt="item.title"/>'
});

const app = new Vue({
  el: '#app',
  data: {
    imageList: [],
    message: 'Hello World Cup!'
  }
});

module.exports = {
  app
};
