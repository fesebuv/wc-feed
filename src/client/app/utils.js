const querystring = require('querystring');

function hasScrolled () {
  if (window && document) {
    return (window.innerHeight + window.pageYOffset) >= document.body.offsetHeight;
  }
  return false;
}

function getUri (page = 0) {
  const params = {
    page
  };
  return `/feed?${querystring.stringify(params)}`;
}

module.exports = {
  hasScrolled,
  getUri
};
