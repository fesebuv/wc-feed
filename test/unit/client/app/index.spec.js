import Vue from 'vue';
import { shallow } from 'vue-test-utils'
import app from '../../../../src/client/app';

describe('app', () => {
  describe('should render markup', () => {
    let component;
    let imageList = [{
      id: '123',
      title: 'an image',
      url_o: 'img src original',
      url_n: 'img src'
    }];

    beforeEach(() => {
      component = shallow(app, {
        data: {
          imageList
        }
      });
    });

    test('data should match imageList', () => {
      expect(component.vm.imageList).toEqual(imageList)
    });

    test('has the expected html structure', () => {
      expect(component.element).toMatchSnapshot();
    });
  });
});
