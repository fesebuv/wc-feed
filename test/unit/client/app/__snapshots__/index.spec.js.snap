// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app should render markup has the expected html structure 1`] = `<!--function (a, b, c, d) { return createElement(vm, a, b, c, d, true); }-->`;
